presto-tests/src/test/java/io/prestosql/execution/TestDiagnosisWarning.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.Session;
+import io.prestosql.execution.warnings.DefaultWarningCollector;
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.execution.warnings.WarningCollectorConfig;
+import io.prestosql.plugin.tpch.TpchConnectorFactory;
+import io.prestosql.spi.PrestoWarning;
+import io.prestosql.spi.WarningCode;
+import io.prestosql.spi.connector.StandardWarningCode;
+import io.prestosql.sql.planner.LogicalPlanner;
+import io.prestosql.testing.LocalQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.fail;
+
+public class TestDiagnosisWarning
```

Comment:
We may want to add `@Test(singleThreaded = true)` and remove redundant `queryRunner` from `assertPlannerWarnings` method.